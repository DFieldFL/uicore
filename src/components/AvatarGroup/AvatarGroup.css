.stack {
  display: inline-flex;
  flex-direction: row-reverse;
}
.noStack {
  display: inline-flex;
}
.stackAvatar:not(:last-child) {
  margin-left: -13px;
}
