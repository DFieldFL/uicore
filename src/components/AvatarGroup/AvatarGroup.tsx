import React from 'react'
import { AvatarProps, Avatar, AvatarSizes } from '../Avatar/Avatar'
import css from './AvatarGroup.css'
import { Color } from '../../core/Color'
import classnames from 'classnames'

export interface AvatarGroupProps {
  onAdd?: (e: React.MouseEvent<HTMLDivElement>) => void
  avatars: Omit<AvatarProps, 'size'>[]
  overlap?: boolean
  avatarGroupProps?: Omit<AvatarProps, 'size'>
  size?: AvatarSizes
}

export const AvatarGroup: React.FC<AvatarGroupProps> = ({
  onAdd,
  avatars,
  overlap = true,
  avatarGroupProps,
  size = 'normal'
}) => {
  if (onAdd) {
    avatars.push({ name: '+', color: Color.BLUE_500, backgroundColor: Color.GREY_200, onClick: onAdd })
  }
  if (overlap) {
    /* in order to create a stack like effect where first avatar overlaps second and second the third one and so on,
    we used flex-direction: row-reverse; but this makes the avatars order from right to left and first avatar will
    be the last one to display from left to right, to correct this we are reversing the array elements
    */
    avatars = avatars.reverse()
  }

  return (
    <div className={classnames(overlap ? css.stack : css.noStack)}>
      {avatars.map(avatarProps => (
        <Avatar
          key={Object.values(avatarProps).join('')}
          className={classnames(overlap && css.stackAvatar)}
          {...avatarGroupProps}
          {...avatarProps}
          {...{ size }}
        />
      ))}
    </div>
  )
}
