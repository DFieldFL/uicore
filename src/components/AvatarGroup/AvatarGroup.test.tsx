import React from 'react'
import { render } from '@testing-library/react'
import { AvatarGroup } from './AvatarGroup'
describe('Render basic component', () => {
  test('should check snapshot with Avatar props with add button', () => {
    const { container } = render(
      <AvatarGroup
        onAdd={() => {}}
        avatars={[
          { name: 'Aahn Boe SS', color: 'red' },
          { email: 'CohnDoe.com.HD@harnes.io', color: 'red' },
          { name: 'Eahn Foe', color: 'red' },
          { name: 'Gahn Hoe', color: 'red' },
          {
            src:
              'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg',
            color: 'red'
          }
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
  test('should check snapshot with Avatar props without add button', () => {
    const { container } = render(
      <AvatarGroup
        avatars={[
          { name: 'Aahn Boe SS', color: 'red' },
          { email: 'CohnDoe.com.HD@harnes.io', color: 'red' },
          { name: 'EahnFoe', color: 'red' },
          { name: 'Gahn Hoe', color: 'red' },
          {
            src:
              'https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg',
            color: 'red'
          }
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
