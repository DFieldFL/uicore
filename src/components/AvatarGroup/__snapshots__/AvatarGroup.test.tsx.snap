// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render basic component should check snapshot with Avatar props with add button 1`] = `
<div>
  <div
    class="stack"
  >
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        +
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        <img
          alt=""
          class="imageStyle"
          src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
          style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
        />
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        GH
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        EF
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        CH
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        AS
      </div>
    </div>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with Avatar props without add button 1`] = `
<div>
  <div
    class="stack"
  >
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        <img
          alt=""
          class="imageStyle"
          src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
          style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
        />
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        GH
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        E
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        CH
      </div>
    </div>
    <div
      class="stackAvatar Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        AS
      </div>
    </div>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with Avatar props without overlap 1`] = `
<div>
  <div
    class="noStack"
  >
    <div
      class="Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        AS
      </div>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        CH
      </div>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        E
      </div>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        GH
      </div>
    </div>
    <div
      class="Avatar contentStyle"
    >
      <div
        class="AvatarInner"
        style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
      >
        <img
          alt=""
          class="imageStyle"
          src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
          style="border-radius: 100%; line-height: 28px; width: 28px; height: 28px; font-size: 10px;"
        />
      </div>
    </div>
  </div>
</div>
`;
